pkg/model/device.go

```diff
@@ -78,16 +75,25 @@ func (d *Device) ToProto() *octopuspb.Device {
 		protoDev.SiteName = d.Site.Name
 	}
 
-	for _, iface := range d.Interfaces {
-		protoDev.Interfaces = append(protoDev.Interfaces, iface.ToProto())
+	if len(d.Interfaces) > 0 {
+		protoDev.Interfaces = make([]*octopuspb.Interface, 0)
```

Comment:
can we use the len(d.Interfaces) to define the len of the list?

Reply:
As the `for` loops are iterating map items rather than items of a list, we need to use `append()`, so we need to create the list with `0` elements.